server/src/main/java/org/elasticsearch/common/util/Maps.java

```diff
@@ -281,6 +282,18 @@ public static <K, V> Map<K, V> newHashMapWithExpectedSize(int expectedSize) {
         return new HashMap<>(capacity(expectedSize));
     }
 
+    /**
+     * Returns a concurrent hash map with a capacity sufficient to keep expectedSize elements without being resized.
+     *
+     * @param expectedSize the expected amount of elements in the map
+     * @param <K> the key type
+     * @param <V> the value type
+     * @return a new pre-sized {@link HashMap}
+     */
+    public static <K, V> Map<K, V> newConcurrentHashMapWithExpectedSize(int expectedSize) {
+        return new ConcurrentHashMap<>(capacity(expectedSize));
```

Comment:
I believe this is unnecessary. Unlike HashMap, ConcurrentHashMap does this calculation in its constructor `long size = (long)(1.0 + (long)initialCapacity / loadFactor);`